### PR TITLE
cleanup ICP code removal

### PR DESCRIPTION
**Pull request by vmamidi**
remove the missed ICP related code.

### CODE REVIEW

File: iocore/net/P_Connection.h [hunk 1]
```diff
@@ -114,7 +114,6 @@ struct Connection {
               );
 
   /// Set the internal socket address struct.
-  /// @internal Used only by ICP.
   void
   setRemote(sockaddr const *remote_addr ///< Address and port.
```
Comment (PSUdaemon) [inline iocore/net/P_Connection.h]: Shouldn't we remove this if it's only used by ICP?

Comment (vmamidi) [inline iocore/net/P_Connection.h]: its used at other places for example Socks.cc

Comment (PSUdaemon) [inline iocore/net/P_Connection.h]: Fair enough.

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/513/

**Comment by atsci** [comment]
AU check *successful*! https://ci.trafficserver.apache.org/job/autest-github/500/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/629/

**Comment by atsci** [comment]
Linux build *successful*! https://ci.trafficserver.apache.org/job/linux-github/2092/

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/504/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/2196/

**Comment by atsci** [comment]
clang-analyzer build *successful*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/757/

**Pull request merged**